library/Vanilla/Models/SSOInfo.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Models;
+
+use Vanilla\ArrayAccessTrait;
+
+/**
+ * Class SSOInfo
+ *
+ * This is a data object containing information returned by an SSOAuthenticator.
+ * Adding methods to this object probably means that you are doing something wrong.
+ *
+ * @package Vanilla\Models
+ */
+class SSOInfo implements \ArrayAccess {
+    use ArrayAccessTrait;
+
+    /**
+     * Maps to "GDN_UserAuthenticationProvider.AuthenticationSchemeAlias"
+     *
+     * @var string
+     */
+    public $authenticatorName;
+
+    /**
+     * Maps to "GDN_UserAuthenticationProvider.ProviderKey"
+     *
+     * @var string
+     */
+    public $authenticatorID;
+
+    /**
+     * Whether the authenticator can sync Roles or User info.
+     *
+     * @var bool
+     */
+    public $authenticatorIsTrusted;
+
+    /**
+     * Maps to "GDN_UserAuthentication.ForeignUserKey"
+     *
+     * @var string
+     */
+    public $uniqueID;
+
+    /**
+     * SSOInfo constructor.
+     *
+     * @param array $associativeArray
+     * @throws \Exception If a non associative array is passed to the constructor.
+     */
+    public function __construct(array $associativeArray = []) {
+        if (array_key_exists(0, $associativeArray)) {
+            throw new \Exception(__CLASS__.' can only be initialized with an associative array.');
+        }
+
+        foreach($associativeArray as $name => $value) {
+            $this->$name = $value;
+        }
+    }
+
+    /**
+     * Getter with default value.
+     * Deprecate when we can use the coalesce operator.
+     *
+     * @param $name
+     * @param $default
+     * @return mixed The extra information or the default if not found.
+     */
+    public function coalesce($name, $default = null) {
```

Comment:
Can you use `get` here, instead? That's pretty much the standard in Vanilla.

Reply:
This object is meant to be used like an array and this function name does exactly what it says. (It is clearer than "get")
It will be removed when we stop supporting php 5.6 (early next year)